demoproviders/barbican/java/org/openyolo/demoprovider/barbican/storage/CredentialStorage.java

```diff
@@ -486,73 +457,65 @@ private File getAuthDomainDir(String authDomain) {
     }
 
     private File initAuthDomainDir(String authDomain) throws IOException {
-        File authorityDir = getAuthDomainDir(authDomain);
-        if (!authorityDir.exists()) {
-            if (!authorityDir.mkdirs()) {
+        File authDomainDir = getAuthDomainDir(authDomain);
+        if (!authDomainDir.exists()) {
+            if (!authDomainDir.mkdirs()) {
                 throw new IOException("failed to create authority dir");
             }
         }
 
-        return authorityDir;
+        return authDomainDir;
     }
 
     private void deleteAuthDomainDir(String authDomain) throws IOException {
-        File credsFile = getCredentialsFile(authDomain);
-        if (credsFile.exists()) {
-            if (!credsFile.delete()) {
-                throw new IOException("failed to delete credentials file");
-            }
+        File authDomainDir = getAuthDomainDir(authDomain);
+        if (!authDomainDir.exists()) {
+            return;
         }
 
-        File authorityDir = getAuthDomainDir(authDomain);
-        if (authorityDir.exists()) {
-            if (!authorityDir.delete()) {
-                throw new IOException("failed to delete authority directory");
-            }
+        boolean success = true;
+        for (File credentialFile : authDomainDir.listFiles()) {
```

Comment:
Unfortunately it's not a recursive delete; you have to delete the individual files.